### PR TITLE
docs: add FAQ page

### PR DESCRIPTION
**Pull request by connorsmith256**
This adds content to the FAQ page, and links to it from the "what is wasmCloud" intro page

Feedback extra welcome on this one. What common questions did I miss?

### CODE REVIEW

File: docs/reference/faq.mdx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+Not exactly. We sometimes refer to wasmCloud [hosts](/docs/concepts/hosts) as "application runtimes." Hosts do run [WebAssembly components](/docs/concepts/webassembly-components), through the use of [wasmtime](https://wasmtime.dev).
+
+### What is the unit of compute for wasmCloud?
+
+wasmCloud applications are composed of [actors](/docs/concepts/actors) and [capability providers](/docs/concepts/capabilities), which are [linked](/docs/concepts/runtime-linking) together at runtime. Decoupling business logic from capabilities is part of the design philosophy of wasmCloud.
+
+### I see wasmCloud actors are stateless, but I need to store state. Is wasmCloud right for me?
+
+wasmCloud applications can access and store state via [capability providers](/docs/concepts/capabilities). Only actors are stateless.
```
Comment (vados-cosmonic) [inline docs/reference/faq.mdx]: Do we want to mention in-memory here? Technically actors can store ephemeral state there right?

File: docs/reference/faq.mdx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+### Do capability providers implement capabilities, or do they act as proxies to external capabilities?
+
+That depends on the specific capability provider. For example, the [HTTP Server provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/providers/http-server) _is_ a server, while the [Redis provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/providers/kv-redis) acts as a client to an external Redis server.
```
Comment (vados-cosmonic) [inline docs/reference/faq.mdx]:
```suggestion
Capability providers can both implement capabilities directly and act as proxies to external capabilities, depending on the needs of your application.

For example, the [HTTP Server provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/providers/http-server) _is_ a server, while the [KeyValue ("KV") Redis provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/providers/kv-redis) acts as a client to an external Redis server.
```

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket:

File: docs/reference/faq.mdx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+### How does wasmCloud handle secrets?
+
+There are several kinds of secrets involved in distributed applications. Configuration for actors and providers can contain sensitive data. Today, these values are associated with [runtime links](/docs/concepts/runtime-linking), or given to providers [on startup](/docs/cli/ctl#options-3). The host itself is also configured with sensitive data, which are passed to the host [on startup](/docs/reference/host-config).
+
+### What is NATS? Why is wasmCloud tightly coupled to NATS?
+
+[NATS](https://nats.io) is open-source "message oriented middleware", responsible for "[addressing, discovery and exchanging of messages that drive the common patterns in distributed systems](https://docs.nats.io/nats-concepts/overview)."
```
Comment (lhmzhou) [inline docs/reference/faq.mdx]: Would it be beneficial to spell out what NATS stand for?

File: docs/developer/debugging/rpc-
Comment (brooksmtownsend) [inline docs/developer/debugging/rpc-:1]: Is this a fake file?

**Review by lhmzhou — COMMENTED** [review]
Looks like you covered a wide range of questions --- nice work!

It'll be extra cool to see answers for: 
1. Can wasmCloud be used with existing container orchestration systems like Kubernetes?
2. What programming languages are supported for writing wasmCloud actors?
3. Are there any performance considerations when using wasmCloud?
4. How does wasmCloud handle versioning of actors and capability providers?
5. What is the typical learning curve for developers new to wasmCloud?
6. How does wasmCloud support logging and monitoring of applications?

File: docs/reference/faq.mdx (hunk 1 above)
Comment (brooksmtownsend) [reply docs/reference/faq.mdx]: If you mean like accessing a global variable, I _think_ we discard all in-memory information after each invocation ends so nothing would carry over.

File: docs/reference/faq.mdx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+wasmCloud applications can access and store state via [capability providers](/docs/concepts/capabilities). Only actors are stateless.
+
+### Do capability providers implement capabilities, or do they act as proxies to external capabilities?
+
+That depends on the specific capability provider. For example, the [HTTP Server provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/providers/http-server) _is_ a server, while the [Redis provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/providers/kv-redis) acts as a client to an external Redis server.
+
+### How does wasmCloud handle secrets?
+
+There are several kinds of secrets involved in distributed applications. Configuration for actors and providers can contain sensitive data. Today, these values are associated with [runtime links](/docs/concepts/runtime-linking), or given to providers [on startup](/docs/cli/ctl#options-3). The host itself is also configured with sensitive data, which are passed to the host [on startup](/docs/reference/host-config).
```
Comment (brooksmtownsend) [inline docs/reference/faq.mdx:75]: Pending a secrets management RFC, I think it's worth calling out application level secrets.
```suggestion
There are several kinds of secrets involved in distributed applications. Configuration for actors and providers can contain sensitive data. Today, these values are associated with [runtime links](/docs/concepts/runtime-linking), or given to providers [on startup](/docs/cli/ctl#options-3). The host itself is also configured with sensitive data, which are passed to the host [on startup](/docs/reference/host-config).

If you're working with secrets in your application, it's recommended to use a capability provider like [kv-vault](https://github.com/wasmCloud/capability-providers/tree/main/kv-vault) that proxies to a secrets management solution.
```

File: docs/reference/faq.mdx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+### What is NATS? Why is wasmCloud tightly coupled to NATS?
```
Comment (brooksmtownsend) [inline docs/reference/faq.mdx]:
```suggestion
### What is NATS? Why does wasmCloud depend on NATS?
```

This just feels better to me but feel free to discard

**Review by brooksmtownsend — COMMENTED** [review]
Only nits, this is a wonderful page 🌮

File: docs/reference/faq.mdx (hunk 1 above)
Comment (connorsmith256) [reply docs/reference/faq.mdx]: Right, actor invocations get fresh state on every invocation. This is by design, so wasmCloud doesn't need to know where to route a particular request; any instance is viable (though NATS leafs will favor local subscribers)

File: docs/developer/debugging/rpc-
Comment (connorsmith256) [reply docs/developer/debugging/rpc-:1]: Super fake

File: docs/reference/faq.mdx (hunk 3 above)
Comment (connorsmith256) [reply docs/reference/faq.mdx]: Today I learned NATS actually stands for something! https://docs.nats.io/reference/faq#what-does-the-nats-acronym-stand-for 

I don't think wasmCloud users need to know this though

**Comment by lachieh** [comment]
Other Questions
- What is a good architecture for a wasmCloud app? (or some other phrasing that allows the response to talk about the Actor Model)
- How do I host frontend applications on wasmCloud? (mention blobby but say other CDN solutions like netlify/vercel are good options)

(these don't need to be added before merge)

**Pull request merged**